sklearn/datasets/samples_generator.py

```diff
@@ -614,19 +614,22 @@ def make_circles(n_samples=100, shuffle=True, noise=None, random_state=None,
     if factor > 1 or factor < 0:
         raise ValueError("'factor' has to be between 0 and 1.")
```

Comment:
do you want to add a test for this line, too? ;)

Reply:
I actually never understood, why a ```factor > 1``` raises a ```ValueError```. All it would to is turn the inner circle into the outer one, allowing me to generate the samples more intuitively. Compare "the one circle's radius is 0.142 of the other one's" vs. "Bigger circler is 7 times the size of the smaller one".

It would remove the dataset's property to be (roughly, due to noise) within a [-1, 1]² box, though...

Reply:
I think leave it as is. Test if you will

Reply:
@christianbraune79 
I think `0 < factor < 1` might be enough. From my perspective, you can use factor to specify how much bigger is the big circle and apply operations on X to specify the position of the big circle (and the small circle). For the example you mentioned, you might use ``X, y = make_circle(factor=1/7); X=7*X`` to achieve your goal. Even if you allow `factor > 1`, you still can't meet the need of many situations, because the center of two circles are fixed to (0,0) and the radius of one circle is fixed to 1.
Also, as the core dev suggested, it will be better if you can add a test to check the error is raised.